### ISSUE TITLE
jsx-uses-vars rule doesn't work with ES6 destructuring assignment

### ISSUE THREAD

**Issue by roman01la** [yannickcr/eslint-plugin-react#35]
Seems like `jsx-uses-vars` rule can not detect used vars in JSX when declaring them this way:

```javascript
var { A } = Obj; // Error: A is defines but never used
<A /> // Using inside React component
```

**Comment by yannickcr** [comment]
In my tests the error is reported by the `no-unused-vars` rule and not `react/jsx-uses-vars`. Can you confirm ?

Seems a duplicate of #18 (which should be fixed with the next ESLint release)

**Comment by roman01la** [comment]
Oh, really. Sorry then :)
